src/libaktualizr/package_manager/packagemanagerinterface.cc

```diff
@@ -96,18 +96,19 @@ bool PackageManagerInterface::fetchTarget(const Uptane::Target& target, Uptane::
       LOG_WARNING << "Skipping download of target with length 0";
       return true;
     }
-    DownloadMetaStruct ds(target, std::move(progress_cb), token);
+    std::unique_ptr<DownloadMetaStruct> ds = std_::make_unique<DownloadMetaStruct>(target, progress_cb, token);
+    assert(ds);
```

Comment:
This assert doesn't seem useful here. make_unique should throw an exception if it fails.

Even then, assert would no be compiled in release builds and that's exactly where we'd want to catch allocation failures.

Reply:
Absolutely agree, just another habit.